src/parallel/DomainDecomposition.cpp

```diff
@@ -32,8 +34,18 @@ void DomainDecomposition::initMPIGridDims() {
 		}
 	}
 
-	MPI_CHECK(MPI_Dims_create( _numProcs, DIMgeom, (int *) &_gridSize ));
-	MPI_CHECK(MPI_Cart_create(MPI_COMM_WORLD, DIMgeom, _gridSize, period, reorder, &_comm));
+	int cStyleGridSize[DIMgeom];
+	for(int i = 0; i < DIMgeom; i++){
+		cStyleGridSize[i] = _gridSize[i];
+	}
+
+	MPI_CHECK(MPI_Dims_create( _numProcs, DIMgeom, (int *) &cStyleGridSize ));
+	MPI_CHECK(MPI_Cart_create(_comm, DIMgeom, cStyleGridSize, period, reorder, &_comm));
```

Comment:
why do you need the C-Style array? Can't you simply use `cStyleGridSize.data()`?

Reply:
I forgot about that, my bad